app/src/main/java/com/ustwo/boilerplate/base/BaseActivity.java

```diff
@@ -0,0 +1,32 @@
+package com.ustwo.boilerplate.base;
+
+import android.os.Bundle;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+public abstract class BaseActivity<V extends BaseView> extends AppCompatActivity {
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(getLayoutId());
+    getPresenter().onViewAttached(getPresenterView());
+  }
+
+  @Override
+  protected void onDestroy() {
+    getPresenter().onViewDetached();
```

Comment:
If you later run Rx unsubscriptions in `onViewDetached` could it be bad to not call from `onPause` instead (with `onViewAttached` called from `onResume`)? Activity could be hidden with expensive observables still subscribed.